cmd/generate.go

```diff
@@ -166,6 +167,20 @@ func (gc *generateCmd) loadAPIModel(cmd *cobra.Command, args []string) error {
 		prop.CertificateProfile.CaPrivateKey = string(caKeyBytes)
 	}
 
+	// If no keypair has been supplied, generate one.
+	if gc.containerService.Properties.LinuxProfile != nil && (gc.containerService.Properties.LinuxProfile.SSH.PublicKeys == nil ||
+		len(gc.containerService.Properties.LinuxProfile.SSH.PublicKeys) == 0 ||
+		gc.containerService.Properties.LinuxProfile.SSH.PublicKeys[0].KeyData == "") {
+		translator := &i18n.Translator{
+			Locale: gc.locale,
+		}
+		var publicKey string
+		_, publicKey, err = helpers.CreateSaveSSH(gc.containerService.Properties.LinuxProfile.AdminUsername, gc.outputDirectory, translator)
```

Comment:
what if `gc.containerService.Properties.LinuxProfile` is nil?

Reply:
then no validation would fail but no key would be created either which would make the cluster deployment fail